File: FMAplication/ClientApp/src/app/Shared/Entity/Inventory/StockAddTransaction.ts
import { PosmProduct } from '..';
import { Transaction } from './Transaction';

export class StockAddTransaction {
    public id: number;
    public transactionId: number;  
    public transactionModel:Transaction;
    public posmProductId: number;  
    public posmProductModel: PosmProduct;
    public quantity: number;  
    public supplier: string;
}

File: FMAplication/Domain/Transaction/StockAddTransaction.cs
using FMAplication.Core;
using FMAplication.Domain.Products;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FMAplication.Domain.Transaction
{
    public class StockAddTransaction : AuditableEntity<int>
    {
        public int TransactionId { get; set; }
        public Transaction Transaction { get; set; }
        public int PosmProductId { get; set; }
        public POSMProduct PosmProduct { get; set; }
        public int Quantity { get; set; }
        public string Supplier { get; set; }
    }
}
